proposals/NNNN-unlock-existentials-with-fixed-associated-types.md

```diff
@@ -0,0 +1,237 @@
+# Unlock Existentials with Fixed Associated Types
+
+* Proposal: [SE-NNNN](NNNN-unlock-existentials-with-fixed-associated-types.md)
+* Authors: [Filip Sakel](https://github.com/filip-sakel), [Anthony Latsis](https://github.com/AnthonyLatsis)
+* Review Manager: TBD
+* Status: **Awaiting implementation**
+
+## Introduction
+
+Today, Swift protocols are divided into two categories: those that _can_ be used as fully-fledged types, and those that can only be used as generic constraints because they have `Self` or associated type requirements. However, in some cases where a protocol inherits and fixes the associated types of another protocol - that doesn't have `Self` requirements, this constraint seems rather unintuitive. This proposal aims to relax this needless contraint allowing such protocols to be used as Types.
+
+## Motivation
+
+Currently, any protocol that has associated type requirements is not allowed to be used as a Type. That's a sensible constraint; however, considering that the associated types of some protocols are known, this restriction can become quite frustrating: 
+
+```swift
+protocol User: Identifiable 
+    where ID == String {
+    
+    var username: String { get }
+    var displayName: String { get }
+}
+
+extension User {
+    var id: String {
+        username
+    }
+}
+```
+
+Many would rightfully assume that `User` could be used as a Type, since we specify that the `ID` asscoated type inherited from [`Identifiable`](https://github.com/apple/swift-evolution/blob/master/proposals/0261-identifiable.md) is `String`. 
+
+```swift
+let myUser: User
+// ❌ `User` is not
+// usable as a Type
+```
```

Comment:
I think we can incorporate this into the `Identifiable` example and simply write ```let myUser: User // error``` (or perhaps we should include the actual error message that compiler emits today).

Reply:
I've incorporated the example into the above code block. 

As for the error message, I intentionally chose to use this "fake" message. It's just for simplicity's sake. I think the readers of Swift Evolution Proposal will have encountered this error before and will, therefore, understand why this code is invalid without being shown the error currently omitted by the compiler.

What do you think? Maybe I'm overestimating how frequent this error is. In my experience this comes up so often that should I mindlessly write such code, I wouldn't even need to read the error message, I would just know.

Reply:
> What do you think? Maybe I'm overestimating how frequent this error is. In my experience this comes up so often that should I mindlessly write such code, I wouldn't even need to read the error message, I would just know.

I agree, `\\ error` should be enough, but you are free to keep the fancy version if you want to.

Reply:
I think we should go for the error message that's kind of descriptive, but not too much. The existing message simply describes that `myUser` is not usable as a Type without getting too technical or detailed.